sunpy/util/tests/test_metadata.py

```diff
@@ -0,0 +1,247 @@
+import pytest
+
+from sunpy.util.metadata import MetaDict
+
+
+def check_contents(metadict_inst, expected):
+    assert len(metadict_inst) == len(expected)
+
+    for key, val in expected:
+        assert metadict_inst[key.upper()] == val
+
+
+def check_insertion_order(metadict_inst, expected):
+
+    def normalise_keys(lst_pairs):
+        return [key.lower() for key, _ in lst_pairs]
+
+    assert normalise_keys(metadict_inst.items()) == normalise_keys(expected)
+
+
+def check(metadict_inst, expected):
```

Comment:
Will restructure tests cases and provide comments better describing 
test intent.

I'll also rename, and document, supporting test functions.